src/ralph/backends/data/base.py

```diff
@@ -473,3 +453,45 @@ async def close(self) -> None:
         Raise:
             BackendException: If a failure occurs during the close operation.
         """
+
+
+def get_backend_generic_argument(
+    backend_class: Type[Union[BaseDataBackend, BaseAsyncDataBackend]], position: int
+) -> Optional[Type]:
+    """Return the generic argument of `backend_class` at specified `position`."""
+    if not hasattr(backend_class, "__orig_bases__"):
+        return None
+
+    bases = backend_class.__orig_bases__[0]
+    if not hasattr(bases, "__args__") or len(bases.__args__) < abs(position) + 1:
+        return None
+
+    argument = bases.__args__[position]
```

Comment:
Why not using `typing.get_args` here? (https://docs.python.org/3/library/typing.html#typing.get_args)

Reply:
Good point) This was [previously mentioned](https://github.com/openfun/ralph/pull/501#discussion_r1399475889) - we did this change in [PR 527](https://github.com/openfun/ralph/pull/527/files#diff-abc4154695f25bd1399562f7c5e9f27fe9fba44ce29e46d606b46f1871b932a6L450-R461) as we have to re-change this method to support a `query_class` of type `BaseQuery` or `str`)

Reply:
Nice, OK!